underactuated.html

```diff
@@ -1312,20 +1329,65 @@ <h1>The Simple Pendulum</h1>
 
-        <li> For each equilibrium point, determine whether it is stable or unstable.  For each one of the stable equilibria, identify the region of attraction.</li>
+        <li> For each equilibrium point, determine whether it is stable or unstable.  For each one of the stable equilibria, identify the basin of attraction.</li>
 
         <li> Consider an additive term $w$ in the system dynamics: $\dot x = f(x) + w$.  As $w$ ranges from $0$ to $\infty$, determine how many stable and unstable equilibrium points the system has. Support your answer with a sketch of the bifurcation diagram for nonnegative values of $w$.</li>
 
       </ol>
 
+      (A note on the terminology: when there is no need to distinguish between different notions of stability, we simply classify equilibrium points as "stable" or "unstable.")
+
+    </exercise>
+
+    <exercise><h1>Enumerate Unstable Equilibria</h1>
+
+      We are given a dynamical system $\dot x = f(x)$ with a scalar state $x$.  The dynamics $f(x)$ is unknown, but we are given two pieces of information:
+
+      <ul>
+
+        <li>the function $f(x)$ is continuous;</li>
+
+        <li>the system has exactly three stable equilibrium points, all of them stable i.s.L.</li>
+
+      </ul>
+
+      Identify the minimum $n_{\text{min}}$ and maximum $n_{\text{max}}$ number of unstable equilibria that the system can have.  Support your answer with the sketch of two functions $f_{\text{min}}(x)$ and $f_{\text{max}}(x)$ that verify the requirements above and have, respectively, $n_{\text{min}}$ and $n_{\text{max}}$ unstable equilibria.
+
+    </exercise>
+
+    <exercise><h1>Pendulum with Vibrating Base</h1>
+
+      Consider an actuated pendulum whose base (pivot of the rod) is forced to oscillate horizontally according to the harmonic law $h \sin(\omega t)$, where $h$ denotes the amplitude of the oscillation and $\omega$ the frequency.  The equation of motion for this system is $$m l^2 \ddot \theta + m g l \sin \theta = m h l \omega^2 \sin (\omega t) + u.$$  (If this equation seems obscure to you, try to derive it using the <a href="underactuated.html?chapter=multibody">Lagrangian approach</a>; but be careful, the kinetic energy $T(t, \theta, \dot \theta)$ of this system depends explicitly on time.)  We would like to design a control law $u (\theta, \dot \theta)$ that makes the pendulum spin at constant velocity $\dot \theta = 1$.  You can do this in two steps:
+
+      <ol type="a">
+
+        <li>come up with a desired closed-loop dynamics $\ddot \theta = f(\dot \theta)$ whose unique equilibrium point is stable and is located in $\dot \theta = 1$;</li>
+
+        <li>use feedback linearization to design a control law $u (\theta, \dot \theta)$ which makes the closed-loop dynamics coincide with the one chosen at the previous point.</li>
+
+        <todo>Add notebook to implement the controller. If possible, with pendulum from urdf and matplotlib animation.</todo>
```

Comment:
totally possible now.  in fact the cartpole urdf is almost this.  you can pull the visualization part of the carpole example off and modify it as necessary.